Modules/Sources/NetworkingCore/Model/MetaDataArray+FlexibleDecoding.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+/// Extension to support flexible decoding of MetaData arrays
+/// Handles both standard array format and object format keyed by index strings
+extension Array where Element == MetaData {
+
+    /// Custom decoding from container that supports both array and dictionary formats
+    public static func decodeFlexibly<K>(from container: KeyedDecodingContainer<K>,
+                                         forKey key: KeyedDecodingContainer<K>.Key) -> [MetaData] {
+        // Try to decode as array first (standard format)
+        if let metaDataArray = try? container.decode([MetaData].self, forKey: key) {
+            return metaDataArray
+        }
+
+        // Try to decode as object keyed by index strings
+        if let metaDataDict = try? container.decode([String: MetaData].self, forKey: key) {
```

Comment:
I was checking if a plugin could produce `[Int: MetaData].self`, but it looks unlikely since using numbers as object keys goes against the JSON specs and `json_encode()` would ensure that the keys are strings.